### PR TITLE
Added service_directory_registrations to forwarding_rule resource

### PR DESCRIPTION
**Pull request by richard-mealey**
DESCRIPTION: Added service_directory_registrations to compute-forwarding_rule resource

References to issues:

Fixes https://github.com/hashicorp/terraform-provider-google/issues/10873

fixes [http://b/224815162] DT: (t-g) Forwarding Rule Support for google_compute_forwarding_rule serviceDirectoryRegistrations #MultiMediaLLC
part of [http://b/213945697] (t-g) Forwarding Rule Support for serviceDirectoryRegistrations #MultiMediaLLC

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note: enhancement
compute: added `service_directory_registrations` field to resource `google_compute_forwarding_rule` (beta)
```

### CODE REVIEW

File: mmv1/third_party/terraform/tests/resource_compute_forwarding_rule_test.go.erb [hunk 1]
```diff
@@ -141,13 +168,36 @@ resource "google_compute_forwarding_rule" "foobar" {
...
+
+	<% unless version == 'ga' %>
+		labels = {
+			baz = "qux"
+		}
+		service_directory_registrations {
+			namespace = google_service_directory_namespace.examplens.namespace_id
+			service = google_service_directory_service.examplesvc.service_id
+		}
+		<% end %>
```
Comment (shuyama1) [inline mmv1/third_party/terraform/tests/resource_compute_forwarding_rule_test.go.erb:180]: nit: we use 2 spaces rather than tabs for spacing here.

**Review by shuyama1 — CHANGES_REQUESTED** [review]
The website docs of this resource is generated via mmv1. Therefore we still need to add the fields to mmv1 to generate docs. Here is the link to the `google_compute_forwarding_rule ` resource: https://github.com/GoogleCloudPlatform/magic-modules/blob/03ca8468c34c6b4d8bd4b7d4cbcdb63c5212145b/mmv1/products/compute/api.yaml#L3745

Here is the [guidance](https://github.com/GoogleCloudPlatform/magic-modules#making-changes-to-resources) and the [example](https://github.com/GoogleCloudPlatform/magic-modules/pull/1126/files#diff-fb4f76e7d870258668a3beac48bf164c) on making changes to resources.
Please let me know if you have any questions adding the fields in mmv1. Sorry for the extra work! Thanks!

File: mmv1/third_party/terraform/tests/resource_compute_forwarding_rule_test.go.erb [hunk 2]
```diff
@@ -98,30 +103,52 @@ func TestAccComputeForwardingRule_networkTier(t *testing.T) {
...
+	resource "google_service_directory_service" "examplesvc" {
+		service_id = "%s"
+		namespace  = google_service_directory_namespace.examplens.id
+
+		metadata = {
+			stage  = "prod"
+			region = "us-central1"
+		}
+	}
+	<% end %>
```
[the remark this answers is not in the document]
Comment (richard-mealey) [reply mmv1/third_party/terraform/tests/resource_compute_forwarding_rule_test.go.erb]: Replaced tabs with 2 spaces

File: mmv1/third_party/terraform/tests/resource_compute_forwarding_rule_test.go.erb (hunk 1 above)
Comment (richard-mealey) [reply mmv1/third_party/terraform/tests/resource_compute_forwarding_rule_test.go.erb]: Replaced tabs with 2 spaces

**Comment by richard-mealey** [comment]
> The website docs of this resource is generated via mmv1. Therefore we still need to add the fields to mmv1 to generate docs. Here is the link to the `google_compute_forwarding_rule ` resource:
> 
> https://github.com/GoogleCloudPlatform/magic-modules/blob/03ca8468c34c6b4d8bd4b7d4cbcdb63c5212145b/mmv1/products/compute/api.yaml#L3745
> 
> Here is the [guidance](https://github.com/GoogleCloudPlatform/magic-modules#making-changes-to-resources) and the [example](https://github.com/GoogleCloudPlatform/magic-modules/pull/1126/files#diff-fb4f76e7d870258668a3beac48bf164c) on making changes to resources. Please let me know if you have any questions adding the fields in mmv1. Sorry for the extra work! Thanks!

**Comment by shuyama1** [comment]
The changes in this PR are added in https://github.com/GoogleCloudPlatform/magic-modules/pull/5994. Therefore, I will close this PR.

**Pull request closed without merging**